### PR TITLE
Interpret: Improve error reporting

### LINKED ISSUE #448: Segment fault at /src/api/MainSolver.h:141 and /src/minisat/mtl/Vec.h:70
Hi, for the following non-standard instances, opensmt https://github.com/usi-verification-and-security/opensmt/commit/82c360a5f0bf7c2ec9026acc4e423dfc73775693 (release build) yields `Segmentation fault`.
I'm not sure if they have values.
```
$ opensmt delta.smt2
Segmentation fault (core dumped)
$ opensmt-asan delta.smt2
ASAN:DEADLYSIGNAL
=================================================================
==24107==ERROR: AddressSanitizer: SEGV on unknown address 0x000000000148 (pc 0x561a70d58712 bp 0x7ffd318c0f30 sp 0x7ffd318c0f10 T0)
==24107==The signal is caused by a READ memory access.
==24107==Hint: address points to the zero page.
    #0 0x561a70d58711 in MainSolver::getLastFrame() const /root/opensmt/src/api/MainSolver.h:141
    #1 0x561a70d58754 in MainSolver::isLastFrameUnsat() const /root/opensmt/src/api/MainSolver.h:142
    #2 0x561a70d476e4 in MainSolver::push() /root/opensmt/src/api/MainSolver.cc:48
    #3 0x561a70cf583e in Interpret::push() /root/opensmt/src/bin/Interpret.cc:595
    #4 0x561a70cf191a in Interpret::interp(ASTNode&) /root/opensmt/src/bin/Interpret.cc:325
    #5 0x561a70cfe504 in Interpret::execute(ASTNode const*) /root/opensmt/src/bin/Interpret.cc:1029
    #6 0x561a70cfe70d in Interpret::interpFile(_IO_FILE*) /root/opensmt/src/bin/Interpret.cc:1042
    #7 0x561a70cd318a in main /root/opensmt/src/bin/opensmt.cc:154
    #8 0x7fc8ec9dbc86 in __libc_start_main (/lib/x86_64-linux-gnu/libc.so.6+0x21c86)
    #9 0x561a70cd2549 in _start (/root/opensmt/build/opensmt+0x3a4549)

AddressSanitizer can not provide additional info.
SUMMARY: AddressSanitizer: SEGV /root/opensmt/src/api/MainSolver.h:141 in MainSolver::getLastFrame() const
==24107==ABORTING
$ cat delta.smt2
(push 1)
```
```
$ opensmt delta2.smt2
Segmentation fault (core dumped)
$ opensmt-asan delta2.smt2
ASAN:DEADLYSIGNAL
=================================================================
==24108==ERROR: AddressSanitizer: SEGV on unknown address 0x000000000230 (pc 0x55fd4502dd43 bp 0x7ffe26a24a20 sp 0x7ffe26a24a10 T0)
==24108==The signal is caused by a READ memory access.
==24108==Hint: address points to the zero page.
    #0 0x55fd4502dd42 in vec<PFRef>::size_() const /root/opensmt/src/minisat/mtl/Vec.h:70
    #1 0x55fd45025231 in MainSolver::PushFramesWrapper::size() const /root/opensmt/src/api/MainSolver.h:80
    #2 0x55fd4501482f in MainSolver::pop() /root/opensmt/src/api/MainSolver.cc:56
    #3 0x55fd44fc28c6 in Interpret::pop() /root/opensmt/src/bin/Interpret.cc:607
    #4 0x55fd44fbeae1 in Interpret::interp(ASTNode&) /root/opensmt/src/bin/Interpret.cc:342
    #5 0x55fd44fcb504 in Interpret::execute(ASTNode const*) /root/opensmt/src/bin/Interpret.cc:1029
    #6 0x55fd44fcb70d in Interpret::interpFile(_IO_FILE*) /root/opensmt/src/bin/Interpret.cc:1042
    #7 0x55fd44fa018a in main /root/opensmt/src/bin/opensmt.cc:154
    #8 0x7f07a97a9c86 in __libc_start_main (/lib/x86_64-linux-gnu/libc.so.6+0x21c86)
    #9 0x55fd44f9f549 in _start (/root/opensmt/build/opensmt+0x3a4549)

AddressSanitizer can not provide additional info.
SUMMARY: AddressSanitizer: SEGV /root/opensmt/src/minisat/mtl/Vec.h:70 in vec<PFRef>::size_() const
==24108==ABORTING
$ cat delta2.smt2
(pop 1)
```

### PR DESCRIPTION
**Pull request by aehyvari**
This PR improves the error reporting in Interpret and tries to do a light-weight cleanup pass on the related code. Fixes #448

### CODE REVIEW

File: src/bin/Interpret.cc [hunk 1]
```diff
@@ -257,30 +257,50 @@ void Interpret::interp(ASTNode& n) {
...
             case t_simplify: {
-                sstat status = main_solver->simplifyFormulas();
-                if (status == s_Error)
-                    notify_formatted(true, "Simplify");
+                if (isInitialized()) {
+                    sstat status = main_solver->simplifyFormulas();
+                    if (status == s_Error)
+                        notify_formatted(true, "Simplify");
+                } else {
+                    notify_formatted(true, "Illegal command before set-logic: declare-const");
```
Comment (blishko) [inline src/bin/Interpret.cc:265]:
```suggestion
                    notify_formatted(true, "Illegal command before set-logic: simplify");
```
Not sure what the exact name is, I did not know we have command for simplification :)

**Review by blishko — COMMENTED** [review]
This looks good, I think!
There seems to be one message that does not match the command.

File: src/bin/Interpret.cc (hunk 1 above)
Comment (aehyvari) [reply src/bin/Interpret.cc]: Looks like it's `simplify`.  I have no idea what it does :)

**Review by blishko — APPROVED** [review]
LGTM!

**Pull request merged**